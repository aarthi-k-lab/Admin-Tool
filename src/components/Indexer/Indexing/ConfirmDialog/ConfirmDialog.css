.confirmationDialog {
    color: #4e586e;
    font-weight: 800 !important;
    font-size: 1.5rem !important;
    line-height: 1.5rem !important;
    font-family: Lato;

    .alertTitle {
        color: #4E586E !important;
        font-weight: 700 !important;
        font-size: 1.5rem !important;
        line-height: 1.5rem !important;
    }

    .dialogButtons {
        justify-content: flex-start !important;
        margin-left: 10px;
        border-radius: 5px !important;
        font-size: 1.1rem;
    }

    .saveButton {
        background-color: #596feb !important;
        color: white !important;
        width: 65px;
        height: 25px;
    }

    .cancelButton {
        width: 65px;
        height: 25px;
    }

    .disabledSaveButton {
        width: 65px;
        height: 25px;
    }
}