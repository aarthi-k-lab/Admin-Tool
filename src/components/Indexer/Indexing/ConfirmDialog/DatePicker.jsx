import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { KeyboardDatePicker, MuiPickersUtilsProvider } from '@material-ui/pickers';
import MomentUtils from '@date-io/moment';
import moment from 'moment-timezone';
import * as R from 'ramda';
import { makeStyles } from '@material-ui/core/styles';

const DATE_FORMAT = 'MM-DD-YYYY';
const useStyles = makeStyles({
  dialogroot: {
    '& .MuiDialogActions-root': {
      display: 'none',
    },
    '& .MuiPickersToolbar-toolbar': {
      minHeight: '48px',
      height: '50px',
    },
    '& .MuiPickersModal-dialogRoot': {
      marginTop: '-13%',
      marginLeft: '8%',
    },
  },
});

function BasicDatePicker(props) {
  const {
    disabled, onChange, value, additionalInfo, failureReason,
  } = props;
  let date = null;

  if (!R.isNil(value)) {
    date = moment(value).isValid() ? moment(value).format(DATE_FORMAT) : null;
  }
  const [datePicker, setDatePicker] = useState(date);
  const { disableFuture } = additionalInfo;
  useEffect(() => {
    if (disabled) {
      setDatePicker(null);
    }
  }, [disabled]);
  const onChangeDatePickerHandler = (selectedDate) => {
    if (selectedDate && selectedDate.isValid()) {
      setDatePicker(selectedDate.format(DATE_FORMAT));
    } else {
      date = R.propOr('', '_i', selectedDate);
      setDatePicker(date);
    }
  };

  const onBlur = () => {
    onChange(datePicker);
  };

  const onAccept = (newDate) => {
    const formattedDate = moment(newDate).format(DATE_FORMAT);
    setDatePicker(formattedDate);
    onChange(formattedDate);
  };

  const isError = !R.isNil(failureReason) && !R.isEmpty(R.filter(item => R.equals(item.level, 1),
    failureReason));
  const classes = useStyles();
  return (
    <MuiPickersUtilsProvider
      utils={MomentUtils}
    >
      <KeyboardDatePicker
        autoOk
        DialogProps={{ className: classes.dialogroot }}
        disabled={disabled}
        disableDialogAction
        disableFuture={disableFuture}
        // disableToolbar
        error={isError}
        format={DATE_FORMAT}
        helperText=""
        inputVariant="outlined"
        KeyboardButtonProps={{
          'aria-label': 'change date',
          style: { padding: '0.2rem' },
        }}
        onAccept={onAccept}
        onBlur={onBlur}
        onChange={onChangeDatePickerHandler}
        position="top-start"
        size="small"
        value={datePicker}
      />
    </MuiPickersUtilsProvider>
  );
}

BasicDatePicker.defaultProps = {
  disabled: false,
  additionalInfo: { hasTitle: false, styleName: '', disableFuture: false },
  failureReason: [],
};

BasicDatePicker.propTypes = {
  additionalInfo: PropTypes.shape({
    disableFuture: PropTypes.bool,
    hasTitle: PropTypes.bool,
    styleName: PropTypes.string,
  }),
  disabled: PropTypes.bool,
  failureReason: PropTypes.arrayOf(PropTypes.shape()),
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
};

export default BasicDatePicker;
