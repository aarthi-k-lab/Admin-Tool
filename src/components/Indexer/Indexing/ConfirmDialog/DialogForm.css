.gridPadding {
    padding-top: 10px;
}

.maQuery {
    color: '#4E586E';
    font-weight: 700;
    font-size: 1.1rem;
}