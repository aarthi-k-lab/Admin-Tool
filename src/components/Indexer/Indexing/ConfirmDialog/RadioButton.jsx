import React from 'react';
import {
  FormControl, FormControlLabel, FormLabel, Radio, FormHelperText, RadioGroup,
} from '@material-ui/core';
import {
  arrayOf, any, string, bool, shape, func,
} from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(() => ({
  customRadioSize: {
    '& .MuiFormGroup-root': {
      paddingLeft: '16px',
    },
    '& .MuiSvgIcon-root': {
      width: '12px',
      height: '12px',
    },
    '& .Mui-checked': {
      color: '#5d97f6',
    },
    '& .MuiFormLabel-root': {
      color: '#4E586E',
      fontWeight: '700 !important',
      fontSize: '1.1rem',
      paddingBottom: '4px',
    },
    '& .MuiRadio-root': {
      paddingRight: '5px',
    },
    '& .MuiFormControlLabel-root': {
      marginBottom: '-8px',
    },
    '& .MuiFormControlLabel-label': {
      fontSize: '1.1rem',
      fontWeight: '400',
    },
  },
}));

const RadioButtons = ({
  disabled = false,
  inputRef,
  options,
  title,
  helperText,
  labelPlacement,
  additionalInfo,
  ...rest
}) => {
  const classes = useStyles();
  return (
    <FormControl
      className={`${classes.customRadioSize}`}
      style={additionalInfo.negativeMargin ? { marginTop: '-135px', marginLeft: '25px' } : {}}
    >
      {title && <FormLabel>{title}</FormLabel>}
      <RadioGroup {...rest}>
        {options.map((option, index) => {
          const radioKey = index + 1;
          return (
            <FormControlLabel
              key={radioKey}
              control={<Radio />}
              disabled={option.disabled || disabled}
              inputRef={inputRef}
              label={option.label}
              labelPlacement={labelPlacement || 'end'}
              style={additionalInfo.margin && radioKey === 2 ? { marginTop: '125px' } : {}}
              value={option.value || ''}
            />
          );
        })}
      </RadioGroup>
      {helperText && <FormHelperText>{helperText}</FormHelperText>}
    </FormControl>
  );
};

RadioButtons.defaultProps = {
  disabled: false,
  helperText: '',
  inputRef: () => {},
  labelPlacement: 'end',
  additionalInfo: {},
  title: undefined,
};

RadioButtons.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  additionalInfo: any,
  disabled: bool,
  helperText: string,
  inputRef: func,
  labelPlacement: string,
  options: arrayOf(
    shape({
      label: string.isRequired,
      value: any.isRequired,
      disabled: bool.isRequired,
    }).isRequired,
  ).isRequired,
  title: string,
};

export default RadioButtons;
