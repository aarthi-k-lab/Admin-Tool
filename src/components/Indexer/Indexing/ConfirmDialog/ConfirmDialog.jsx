import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import {
  Button, Dialog, DialogTitle, DialogContent, DialogActions, Typography,
} from '@material-ui/core';
import DialogForm from './DialogForm';
import './ConfirmDialog.css';

const ConfirmationDialog = (props) => {
  const {
    isOpen, handleClose, saveFn = () => {}, cancelFn = () => {}, isMaLoan,
  } = props;
  const defaultValues = {
    mmoSelection: '', mmoDate: '', dispositionOption: '', soiReceived: '', mmoOption: '',
  };
  const [disableSave, setDisableSave] = useState(true);
  const [values, setValues] = useState(defaultValues);

  useEffect(() => {
    if (!isOpen) {
      setValues(defaultValues);
    }
  }, [isOpen]);

  useEffect(() => {
    const {
      dispositionOption, soiReceived, mmoOption, mmoSelection,
    } = values;
    if (dispositionOption && soiReceived) {
      if ((isMaLoan && mmoOption === 'No')
      || (isMaLoan && mmoOption === 'Yes' && mmoSelection)
      || (!isMaLoan && !mmoOption)) {
        setDisableSave(false);
      } else {
        setDisableSave(true);
      }
    } else { setDisableSave(true); }
  }, [values]);

  return (
    <Dialog
      onClose={() => {
        handleClose();
        setValues(defaultValues);
      }}
      open={isOpen}
      PaperProps={{
        style: {
          minWidth: '40%', borderRadius: '10px', backgroundColor: '#FFFFFF',
        },
      }}
      styleName="confirmationDialog"
    >
      <DialogTitle>
        <span styleName="alertTitle">
          {'Confirmation Dialog'}
        </span>
      </DialogTitle>
      <DialogContent dividers>
        <Typography gutterBottom>
          <DialogForm isMaLoan={isMaLoan} setValues={setValues} values={values} />
        </Typography>
      </DialogContent>
      <DialogActions styleName="dialogButtons">
        <Button
          disabled={disableSave}
          onClick={() => saveFn(values)}
          styleName={disableSave ? 'disabledSaveButton' : 'saveButton'}
          variant="contained"
        >
Save

        </Button>
        <Button
          color="secondary"
          disabled={false}
          onClick={() => { cancelFn(); setValues({}); }}
          styleName="cancelButton"
          variant="outlined"
        >
        Cancel

        </Button>
      </DialogActions>
    </Dialog>
  );
};


ConfirmationDialog.defaultProps = {
  handleClose: () => {},
  isMaLoan: false,
};

ConfirmationDialog.propTypes = {
  cancelFn: PropTypes.func.isRequired,
  handleClose: PropTypes.func,
  isMaLoan: PropTypes.bool,
  isOpen: PropTypes.bool.isRequired,
  saveFn: PropTypes.func.isRequired,
};

export default ConfirmationDialog;
