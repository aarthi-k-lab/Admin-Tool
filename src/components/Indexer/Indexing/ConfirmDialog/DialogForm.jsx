import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { Grid, IconButton } from '@material-ui/core';
import RadioButton from './RadioButton';
import {
  DISPOSITION_OPTIONS, SOI_OPTIONS, MMO_OPTIONS, MMO_SELECTIONS,
} from '../../../../constants/indexer';
import DatePicker from './DatePicker';
import './DialogForm.css';

const DialogForm = ({ values, setValues, isMaLoan }) => {
  const [isExpanded, setIsExpanded] = useState(false);
  const [disableDate, setDisableDate] = useState(true);
  const [disableToggle, setDisableToggle] = useState(true);
  const [mmoRadioMargin, setMmoRadioMargin] = useState(false);
  useEffect(() => {
    if (values.mmoOption === 'Yes') {
      setDisableDate(false);
      if (values.mmoDate) {
        setDisableToggle(false);
        setIsExpanded(true);
        setMmoRadioMargin(true);
      } else {
        setDisableToggle(true);
        setIsExpanded(false);
        setMmoRadioMargin(false);
      }
    } else {
      setDisableDate(true);
      setValues(
        prevValues => ({ ...prevValues, mmoDate: '', mmoSelection: '' }),
      );
      setIsExpanded(false);
      setDisableToggle(true);
      setMmoRadioMargin(false);
    }
  }, [values.mmoOption, values.mmoDate]);

  return (
    <>
      <RadioButton
        defaultValue=""
        onChange={(e) => {
          setValues(prevValues => ({ ...prevValues, dispositionOption: e.currentTarget.value }));
        }}
        options={DISPOSITION_OPTIONS}
        title="Please select the Disposition option"
        value={values.dispositionOption || ''}
      />
      <div styleName="gridPadding">
        <RadioButton
          defaultValue=""
          onChange={(e) => {
            setValues(prevValues => ({ ...prevValues, soiReceived: e.currentTarget.value }));
          }}
          options={SOI_OPTIONS}
          title="Was a Statement of Information (SOI) received"
          value={values.soiReceived || ''}
        />
      </div>
      {isMaLoan && (
      <div styleName="gridPadding">
        <Grid container spacing={2}>
          <Grid item lg={12} md={12} sm={12} xl={12} xs={12}>
            <span style={{ color: '#4E586E' }} styleName="maQuery">
Was an Executed Mortgage Modification Options (MMO) form received?
            </span>
          </Grid>
          <Grid item lg={12} md={12} sm={12} xl={12} xs={12}>
            <Grid
              container
              spacing={2}
            >
              <Grid item lg={1} md={1} sm={1} xl={1} xs={1}>
                <RadioButton
                  additionalInfo={{ margin: mmoRadioMargin }}
                  defaultValue=""
                  onChange={(e) => {
                    setValues(prevValues => ({ ...prevValues, mmoOption: e.currentTarget.value }));
                  }}
                  options={MMO_OPTIONS}
                  value={values.mmoOption || ''}
                />

              </Grid>
              <Grid
                item
                lg={3}
                md={3}
                sm={3}
                style={{ marginLeft: '25px' }}
                xl={3}
                xs={3}
              >
                <DatePicker
                  disabled={disableDate}
                  onChange={(e) => {
                    setValues(prevValues => ({ ...prevValues, mmoDate: e }));
                  }}
                  value={values.mmoDate}
                />
              </Grid>
              <Grid item lg={2} md={2} sm={2} style={{ marginLeft: '50px' }} xl={2} xs={2}>
                <IconButton
                  disabled={disableToggle}
                  disableRipple
                  onClick={() => {
                    setIsExpanded((prevState) => {
                      setMmoRadioMargin(!prevState);
                      return !prevState;
                    });
                  }}
                  style={disableToggle ? { opacity: '0.5', padding: '12px' } : { opacity: '1', padding: '12px' }}
                >
                  {isExpanded ? <img alt="collapse" src="/static/img/collapse.svg" />
                    : <img alt="expand" src="/static/img/expand.svg" />}
                </IconButton>
              </Grid>
            </Grid>
          </Grid>
          {isExpanded && !disableToggle && (
          <RadioButton
            additionalInfo={{ negativeMargin: true }}
            defaultValue=""
            onChange={(e) => {
              setValues(prevValues => (
                { ...prevValues, mmoSelection: e.currentTarget.value }));
            }}
            options={MMO_SELECTIONS}
            title="Please select the MMO Selection"
            value={values.mmoSelection || ''}
          />
          )}
        </Grid>
      </div>
      )}
    </>
  );
};

DialogForm.propTypes = {
  isMaLoan: PropTypes.bool.isRequired,
  setValues: PropTypes.func.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  values: PropTypes.any.isRequired,
};
export default DialogForm;
