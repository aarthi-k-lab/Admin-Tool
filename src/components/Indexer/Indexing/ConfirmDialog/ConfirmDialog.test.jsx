import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import ConfirmDialog from './ConfirmDialog';

const defaultProps = {
  isOpen: true,
  handleClose: jest.fn(),
  saveFn: jest.fn(),
  cancelFn: jest.fn(),
  isMaLoan: false,
};

describe('ConfirmDialog', () => {
  describe('ConfirmDialog snapshot', () => {
    const rootReducer = (state = { dashboard: { resultOperation: { status: '' } } }) => state;
    const store = createStore(rootReducer);
    it('should take ConfirmDialog snapshot', () => {
      const component = shallow(
        <Provider store={store}><ConfirmDialog {...defaultProps} /></Provider>,
      );
      expect(component).toMatchSnapshot();
    });
    it('does not show third question for isMaLoan false props', () => {
      const wrapper = shallow(
        <Provider store={store}><ConfirmDialog {...defaultProps} /></Provider>,
      );
      const spanElement = wrapper.find('span.maQuery');
      expect(spanElement.exists()).toBe(false);
    });
  });
});
