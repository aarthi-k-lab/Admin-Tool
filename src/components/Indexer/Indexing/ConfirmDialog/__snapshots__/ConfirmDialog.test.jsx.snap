// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDialog ConfirmDialog snapshot should take ConfirmDialog snapshot 1`] = `ShallowWrapper {}`;
